src/Traits/VerifiableTrait.php

```diff
@@ -15,20 +14,33 @@ trait VerifiableTrait
 
     protected function loadActiveVerifications(string $action)
     {
-        $newVerifications = $this->shouldKeepDuringSession($action)
-            ? VerificationCode::allFor($this)->whereNull('verifies_until')->whereNotNull('used_at')->get()
-            : VerificationCode::allFor($this)->where('verifies_until', '>', Carbon::now()->toDateTime())->get();
+        $newVerifications = $this->getNewVerifications($action);
 
         $this->activeVerifications = is_null($this->activeVerifications)
             ? $newVerifications
             : $this->activeVerifications->merge($newVerifications);
     }
 
-    private function shouldKeepDuringSession(string $action): bool
+    private function getNewVerifications(string $action): Collection
     {
-        return (Config::get('verifications.actions.'.$action.'.keep_verified_during_session') === true)
-            && Session::has('last_activity')
-            && (Session::get('last_activity') < Carbon::now()->addMinutes(Config::get('session.lifetime'))->toDateTime());
+        $now = Carbon::now();
+
+        switch (Config::get('verifications.actions.'. $action .'.expires_from')) {
+            case 'last-activity':
+                 $verificationsQuery = VerificationCode::allActiveForAction($this, $action, $now->toDateTime());
+                 $verificationsQuery->update([
```

Comment:
Pri kazdom nacitani sa rovno updatuje last_touched_at? Alebo kedy sa toto zavola? Iba ak je routa schovana za middlewarom?

Reply:
Toto sa zavola vzdy, ked sa zavola **shouldVerify()**, cize pri kazdej akcii/route, ktora ma byt chranena. a samozrejme k updatovaniu tejto hodnoty dochadza len pre akcie, ktore maju konfiguraciu **expires_from => 'last-activity'**

Reply:
Pride mi to ako magic. Ja by som ocakaval ze budeme mat metodku niekde zadefinovanu `touchVerification()` ktoru budeme explicitne volat v shouldVerify() za splnenia podmienok, cize asi by som shouldVerify() refacorol tak nejak:
```
    /**
     * @param string $action
     * @return bool
     */
    public function shouldVerify(string $action): bool
    {
        if (!Config::get('verifications.enabled') || is_null($this->getUser()) {
            return false;
        }

        // asi taketo nieco ze? otazka ci to ma byt pred isVerificationEnabled
        // alebo to ma byt medzi isVerificationEnabled a isVerificationActive, nad tym sa zamysli prosim ta
        $this->getUser()->touchVerificationIfExists($action); 

        return $this->getUser()->isVerificationEnabled($action)
            && !$this->getUser()->isVerificationActive($action);
    }
```